CDCgov/NEDSS-Modernization
# Refactors Redirect tests to ise ResultActions

## Description

This a refactor to allow changes that will be made for `CNFT1-1559`.  The Search Redirect tests were changed to call endpoints and verify end results over calling services directly.  This will help with `CNFT1-1559` by allowing refactoring of the `PatientSearcher` without having to change the redirect tests in the future

## Tickets

* [CNFT1-1559](https://cdc-nbs.atlassian.net/browse/CNFT1-1559)

## Steps to verify

1. Ensure that all tests pass by running `./gradlew clean test`


[CNFT1-1559]: https://cdc-nbs.atlassian.net/browse/CNFT1-1559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
